### PR TITLE
Don't generate end element on apply completion if it exists

### LINKED ISSUE #651: Don't generate end element on apply completion if it exists
To fix https://github.com/redhat-developer/vscode-xml/pull/213#issuecomment-579328288 the apply completion (for XML and for XML based on a grammar) must generate the end element only if it needs.

### PR DESCRIPTION
**Pull request by angelozerr**
Don't generate end element on apply completion if it exists

Fixes #651

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
To test this PR you need to test XML completion based on XML Schema/DTD and without it and open completion before an orphelan end element.

Create a XML Schema choice.xsd like this:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<xs:schema
        xmlns:xs="http://www.w3.org/2001/XMLSchema"
        elementFormDefault="qualified"
        attributeFormDefault="unqualified">
    <xs:element name="person">
        <xs:complexType>
            <xs:choice
                    minOccurs="1"
                    maxOccurs="2">
                <xs:element name="employee"/>
                <xs:element name="member"/>
            </xs:choice>
        </xs:complexType>
    </xs:element>
</xs:schema>
```

create a XML file like this:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<person xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="choice.xsd">
	<employee></employee>
	|</member>
</person>
```

open completion before member and apply it, it should generate only `<member>` without close it.

After write a XML file (without XSD) like this:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<person>
	<employee></employee>
	|</employee>
</person>
```

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java [hunk 1]
```diff
@@ -378,7 +379,49 @@ public boolean isEndTagClosed() {
 	}
 
 	/**
-	 * If Element has a closing end tag eg: <a> </a> -> true , <a> </b> -> false
+	 * Returns true if the given element is an orphan end tag (which have not a
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java:382]: which has no start tag

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java [hunk 2]
```diff
@@ -378,7 +379,49 @@ public boolean isEndTagClosed() {
 	}
 
 	/**
-	 * If Element has a closing end tag eg: <a> </a> -> true , <a> </b> -> false
+	 * Returns true if the given element is an orphan end tag (which have not a
+	 * start tag, eg: </a>) and false otherwise.
+	 * 
+	 * @param tagName the end tag name.
+	 * @return true if the given element is an orphan end tag (which have not a
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java:386]: has no start tag

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java]: fixed

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java (hunk 2 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/dom/DOMElement.java]: fixed

**Pull request merged**